OpenRA.Mods.Common/Util.cs

```diff
@@ -155,5 +158,70 @@ public static IEnumerable<CPos> RandomWalk(CPos p, MersenneTwister r)
 				yield return p;
 			}
 		}
+
+		public static string FriendlyTypeName(Type t)
+		{
+			if (t.IsGenericType && t.GetGenericTypeDefinition() == typeof(HashSet<>))
+				return "Set of {0}".F(t.GetGenericArguments().Select(FriendlyTypeName).ToArray());
+
+			if (t.IsGenericType && t.GetGenericTypeDefinition() == typeof(Dictionary<,>))
+				return "{0}, {1} pairs".F(t.GetGenericArguments().Select(FriendlyTypeName).ToArray());
+
+			if (t.IsGenericType && t.GetGenericTypeDefinition() == typeof(IList<>)) // TODO: why does this not work?
+				return "List of {0}".F(FriendlyTypeName(t.GetElementType()));
+
+			if (t.IsSubclassOf(typeof(Array)))
+				return "Multiple {0}".F(FriendlyTypeName(t.GetElementType()));
+
+			if (t.IsGenericType && t.GetGenericTypeDefinition() == typeof(Primitives.Cache<,>))
+				return "Cached<{0},{1}>".F(t.GetGenericArguments().Select(FriendlyTypeName).ToArray());
```

Comment:
I tracked this down to https://github.com/OpenRA/OpenRA/commit/7d0f4a261f3f2260fd05fdffebb88080683e768e and it is exposed at https://github.com/OpenRA/OpenRA/wiki/Traits#mobile but it doesn't seem to look broken in the trait documentation so I am not sure if this even fires (anymore).